spec/packets_spec.cr

```diff
@@ -177,7 +177,7 @@ describe MQTT::Protocol::Packet do
           connect = connect.should be_a MQTT::Protocol::Connect
           connect.username.should eq username
           connect.password.should eq password
-          connect.will.not_nil!.topic.should eq wtopic
+          connect.will.try(&.topic).should eq wtopic
```

Comment:
Nowadays, In specs i usually do
```suggestion
         will = connect.will.should_not be_nil
         will.topic.should eq wtopic
```
to get rid of `not_nil!`s. An extra line and assertion, but usually I think that's good thing.